framework/src/packets/icmp/v6/ndp/router_advert.rs

```diff
@@ -199,6 +202,12 @@ impl<E: Ipv6Packet> Icmpv6<E, RouterAdvertisement> {
     pub fn set_retrans_timer(&mut self, retrans_timer: u32) {
         self.payload_mut().retrans_timer = u32::to_be(retrans_timer);
     }
+
+    /// Returns the offset where the data field where the option starts
+    #[inline]
+    pub fn option_offset(&self) -> usize {
+        self.offset + Icmpv6Header::size() + RouterAdvertisement::size()
```

Comment:
not sure if this is the best way to get the option offset. we basically need the icmp_offset + icmpheader size + router advertisement size

Reply:
option offset is not going to help much because it's only the start of where the options are but if you have more than one option, it's probably not where you want to add the next one. you want to just add it to the very end of the buffer. which is `mbuf.data_len()`

Reply:
removed